### PR TITLE
Move turb inflow helpers from PeleLMeX and add documentation

### PR DESCRIPTION
**Pull request by baperry2**
Address the long asked-for need for better documentation of the turbinflow capability

- Brings over turbinflow support codes, which include the python script that creates the isotropic turbulence and the amrex code that repackages it for use as a turbinflow. 
- Adds documentation for the support codes and the PeleLMeX utility that handles turbinflows
- Includes one behavior change that was necessary for the SAF project: overlapping turbinflow patches are added together rather than overwriting each other. (@SreejithNREL please confirm I put the correct change in here: https://github.com/AMReX-Combustion/PelePhysics/pull/489/commits/73f9f12c6f893a9290634a9a101b5863da5f69b4).

### CODE REVIEW

File: Support/TurbFileHIT/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+First generate the data using the python script:
+./gen_hit_ic.py -k0 4 -N 128
+
+To generate a synthetic HIT field discretized with 128 cells and most energetic eddies 
+at a wave number of 4.
+
+Then compile the C++ executable (AMReX needed):
+make
+
+And the executable to generature the turbfile (adapt the input file to your needs):
```
Comment (marchdf) [inline Support/TurbFileHIT/README.md]: generature -> generate

File: Docs/sphinx/Support.rst [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+to provide turbulent inflow boundary conditions in either PeleC or PeleLMeX.
+
+Generating an HIT File
+~~~~~~~~~~~~~~~~~~~~~~
+
+The ``gen_hit_ic.py`` python script creates a synthetic isotropic turbulence velocity field based on the methodology described
+in the appendix of `Johnsen et al. (2010) J. Comp Phys. <http://dx.doi.org/10.1016/j.jcp.2009.10.028>`_. The resulting velocity
+field is stored in files with names following the pattern ``hit_ic_k0_N.dat``, where ``k0`` is the most
+energetic wave number and ``N`` is the size of the grid. The initial
+condition for a grid of size N^3 is generated as follows:
```
Comment (marchdf) [inline Docs/sphinx/Support.rst]: you can `math` the N^3 and other such math expressions below

**Review by marchdf — COMMENTED** [review]
Fantastic work. Thank you!

**Comment by SreejithNREL** [comment]
I checked the commit and I confirm the change is correct

**Review by SreejithNREL — APPROVED** [review]

**Review by marchdf — APPROVED** [review]

**Pull request merged**